lib-impl/src/main/java/org/ergoplatform/appkit/impl/NodeAndExplorerDataSourceImpl.java

```diff
@@ -156,6 +156,12 @@ public InputBox getBoxByIdWithMemPool(String boxId) {
         return new InputBoxImpl(boxData);
     }
 
+    @Override
+    public InputBox getBoxByIdWithSpent(String boxId) {
+        OutputInfo outputInfo = executeCall(explorerApi.getApiV1BoxesP1(boxId));
+        return new InputBoxImpl(outputInfo);
+    }
```

Comment:
What is the difference with getBoxById other than different types involved?

Reply:
`getBoxById` only returns unspent boxes. It would be better to rename it, but that would break a lot of client code.

Reply:
Then maybe give it a more descriptive name or add parameter
`InputBox getBoxById(String boxId, boolean searchInSpentBoxes)`

Reply:
What do you suggest? To be backwards compatible, we need these two:

`getBoxById(String boxId)` -> only unspent boxes
`getBoxById(String boxId, boolean searchInSpentBoxes)` -> either only unspent boxes or search in spent boxes as well

this breaks down into the two options that are available now as well, only that instead of `getBoxByIdWithSpent` you need to call `getBoxById(String boxId, true)`

Reply:
I think the meaning of getBoxById(String boxId, boolean searchInSpentBoxes) is more clear than getBoxByIdWithSpent (that "WithSpent" is not clear). At the same time I don't like overloading methods.
An alternative name `findBoxByIdIncludingSpent`, which suggest that there will be some `find` which includes the spent boxes.

Reply:
Мaybe just one generic method `findBoxById(id, findInPool = true, findInSpent = true)`
or even introduce a class `SearchParameters(findInUtxo: Boolean, findInPool:Boolean, findInSpent: Boolean)` 
With this you can find everywhere except Utxo, which is not possible now.

Reply:
Done with the parameter flags.